src/main/java/mate/academy/mainspringproject/repository/BookRepositoryImpl.java

```diff
@@ -0,0 +1,51 @@
+package mate.academy.mainspringproject.repository;
+
+import java.util.List;
+import mate.academy.mainspringproject.model.Book;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class BookRepositoryImpl implements BookRepository {
+
+    private final SessionFactory sessionFactory;
+
+    @Autowired
+    public BookRepositoryImpl(SessionFactory sessionFactory) {
+        this.sessionFactory = sessionFactory;
+    }
```

Comment:
if we are using Lombok - lets use it
```suggestion
@Repository
@RequiredArgsConstructor
public class BookRepositoryImpl implements BookRepository {

    private final SessionFactory sessionFactory;
```